hnn_core/tests/test_viz.py

```diff
@@ -211,4 +222,117 @@ def test_dipole_visualization():
+        net_plot.export_movie('demo.gif', dpi=200)
+
+    # Errors if vsec isn't recorded with record_vsec='all'
+    _ = simulate_dipole(net, dt=0.5, tstop=10, record_vsec='soma')
+    net_plot = NetworkPlotter(net)
+
+    assert net_plot.vsec_array.shape == (159, 1)
+    assert net_plot.color_array.shape == (159, 1, 4)
+    assert net_plot._vsec_recorded is False
+
+    with pytest.raises(RuntimeError, match='Network must be simulated'):
+        net_plot.export_movie('demo.gif', dpi=200)
+
+    # Simulate with record_vsec='all' to test voltage plotting
+    net = jones_2009_model(params)
+    _ = simulate_dipole(net, dt=0.5, tstop=10, n_trials=2,
+                        record_vsec='all')
+    net_plot = NetworkPlotter(net)
+
+    assert net_plot.vsec_array.shape == (159, 21)
+    assert net_plot.color_array.shape == (159, 21, 4)
+    assert net_plot._vsec_recorded is True
+    assert isinstance(net_plot._cbar, Colorbar)
+
+    # Type check errors
+    with pytest.raises(TypeError, match='xlim must be'):
+        net_plot.xlim = 'blah'
+    with pytest.raises(TypeError, match='ylim must be'):
+        net_plot.ylim = 'blah'
+    with pytest.raises(TypeError, match='zlim must be'):
+        net_plot.zlim = 'blah'
+    with pytest.raises(TypeError, match='elev must be'):
+        net_plot.elev = 'blah'
+    with pytest.raises(TypeError, match='azim must be'):
+        net_plot.azim = 'blah'
+    with pytest.raises(TypeError, match='vmin must be'):
+        net_plot.vmin = 'blah'
+    with pytest.raises(TypeError, match='vmax must be'):
+        net_plot.vmax = 'blah'
+    with pytest.raises(TypeError, match='trial_idx must be'):
+        net_plot.trial_idx = 1.0
+    with pytest.raises(TypeError, match='time_idx must be'):
+        net_plot.time_idx = 1.0
+    with pytest.raises(TypeError, match='colorbar must be'):
+        net_plot.colorbar = 'blah'
+
+    # Check that the setters work
+    net_plot.xlim = (-100, 100)
+    net_plot.ylim = (-100, 100)
+    net_plot.zlim = (-100, 100)
+    net_plot.elev = 10
+    net_plot.azim = 10
+    net_plot.vmin = 0
+    net_plot.vmax = 100
+    net_plot.trial_idx = 1
+    net_plot.time_idx = 5
+    net_plot.bgcolor = 'white'
+    net_plot.voltage_colormap = 'jet'
+
+    net_plot.colorbar = False
+    # check later that net._cbar is None
+    assert net_plot._cbar is None
+
+    # Check that the getters work
+    assert net_plot.xlim == (-100, 100)
+    assert net_plot.ylim == (-100, 100)
+    assert net_plot.zlim == (-100, 100)
+    assert net_plot.elev == 10
+    assert net_plot.azim == 10
+    assert net_plot.vmin == 0
+    assert net_plot.vmax == 100
+    assert net_plot.trial_idx == 1
+    assert net_plot.time_idx == 5
+
+    assert net_plot.bgcolor == 'white'
+    assert net_plot.fig.get_facecolor() == (1.0, 1.0, 1.0, 1.0)
+
+    assert net_plot.voltage_colormap == 'jet'
+
+    # Test animation export and voltage plotting
+    net_plot.export_movie('demo.gif', dpi=200, decim=100, writer='pillow')
```

Comment:
I guess this will test if this operation crashes. A more comprehensive test would be to make sure this file actually exists. By following up with an os.path.exists() or os.path.isfile(). Bonus points for writing to temp space.

Reply:
Borrowed your logic from the network config PR FOR THIS ONE